### PR TITLE
updated user management test cases

### PR DESCRIPTION
**Pull request by priya-badmanji**
updated user management test cases. 
| | |||
|-|-|-|-|
| Authentication | Authenticate | Validation |When invalid user credentials provided | Expect to return HTTP status 401 |
| Authentication | Authenticate | Functional |When valid user credentials provided | Expect to return object of type Token containing access_token & refresh_token ||
| Authentication | Refresh Token | Validation |When provided invalid refresh token | Expect to return HTTP status 500 ||
| Authentication | Refresh Token | Functional |When provided valid refresh token | Expect to return object of type Token containing access_token & refresh_token ||
UserManagement | Assign Permission | Functional | When assigning valid user permission | Expect to return true | 
-- | -- | -- | -- | -- | --
UserManagement | Assign Permission | Auth | When no auth token provided | Expect to return HTTP status 401 | 
UserManagement | Assign Permission | Validation | When invalid username provided | Expect to return HTTP status 404 | 
UserManagement | Assign Permission | Validation | When managing own account permission | Expect to return HTTP status 400 | 
-- | -- | -- | -- | -- | --
UserManagement | Revoke Permission | Functional | When revoking valid user permission | Expect to return true | 
UserManagement | Revoke Permission | Auth | When no auth token provided | Expect to return HTTP status 401 | 
UserManagement | Revoke Permission | Validation | When invalid username provided | Expect to return HTTP status 404 | 
UserManagement | Revoke Permission | Validation | When managing own account permission | Expect to return HTTP status 400 | 
-- | -- | -- | -- | -- | --
UserManagement | User Org Roles | Functional | When fetching logged in user org roles | Expect to return user org roles of type OrgRoles | 
UserManagement | User Org Roles | Auth | When no auth token provided | Expect to return HTTP status 401

### CODE REVIEW

File: src/__tests__/authentication.test.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      });
+    });
+
+    describe("Functional", () => {
+      it("When provided valid refresh token, Expect to return object of type Token containing access_token & refresh_token", async () => {
+        //Arrange
+        let configuration = Utility.getConfiguration();
+        let generalAPI = new AuthApi(configuration);
+        //Act
+        let loginResponse = await generalAPI.authenticate({
```
Comment (susrisha) [inline src/__tests__/authentication.test.ts:67]: in this test, getting login response is still into Arrange part.

File: src/__tests__/user-management.test.ts [hunk 2]
```diff
@@ -199,4 +197,123 @@ describe("User Management service", () => {
...
+        //Assert
+        await expect(orgRoles()).rejects.toMatchObject({ response: { status: 401 } });;
+      });
+    });
+
+    describe("Functional", () => {
+      it("When fetching logged in user org roles, Expect to return user org roles of type OrgRoles", async () => {
+        //Arrange
+        let configuration = Utility.getConfiguration();
+        const loginResponse = await Utility.login(seederData?.user.email!, "Tester01*");
```
Comment (susrisha) [inline src/__tests__/user-management.test.ts:210]: Is it ok to write the password?

File: src/__tests__/user-management.test.ts [hunk 3]
```diff
@@ -199,4 +197,123 @@ describe("User Management service", () => {
...
+        //Act
+        const response = await userManagementApi.revokePermission(<RoleDetails>
+          {
+            roles: [TDEIROLES.FLEX_DATA_GENERATOR],
+            tdei_org_id: seederData?.organizationId,
+            user_name: seederData?.user.email
+          });
+
+        //Assert
+        expect(response.data.data).toBe("Successful!");
```
Comment (susrisha) [inline src/__tests__/user-management.test.ts:292]: can we also assert with the status?

**Review by susrisha — CHANGES_REQUESTED** [review]
One minor change while dealing with rejects. Instead of writing
```js
const login = async ()=>{await generalAPI.authenticate(); }
await expect(login()).rejects.toMatchObject({response:{status:401}});
```
Try using the following
```js
const login = generalAPI.authenticate();
await expect(login).rejects.toMatchObject({response:{status:401}});
```
This applies for all the cases

File: src/__tests__/user-management.test.ts (hunk 2 above)
Comment (priya-badmanji) [reply src/__tests__/user-management.test.ts:210]: Default password for seed user.

File: src/__tests__/user-management.test.ts (hunk 3 above)
Comment (priya-badmanji) [reply src/__tests__/user-management.test.ts:292]: will do

**Review by susrisha — APPROVED** [review]

**Pull request merged**